tiny_dnn/layers/layer.h

```diff
@@ -685,6 +827,25 @@ class layer : public node {
     return true;
   }
 
+  bool has_same_parameters(const layer &rhs, float_t eps) const {
```

Comment:
@karandesai-96 what's the use case of this? I think that will make more sense to overload the bool operator in Tensor.